drivers/block-nbd.c

```diff
@@ -718,92 +675,96 @@ tdnbd_nbd_negotiate_old(struct tdnbd_data *prv, td_driver_t *driver)
 
 	if (rc != 0) {
 		ERROR("Could not set O_NONBLOCK flag");
-		close(sock);
-		return -1;
+		goto errout;
 	}
-	INFO("Successfully connected to NBD server");
+	INFO("Successfully connected to Old-style NBD server");
 
 	return 0;
+
+errout:
+	close(sock);
+	return -1;
 }
 
 static int
 tdnbd_nbd_negotiate_new(struct tdnbd_data *prv, td_driver_t *driver)
 {
 	int rc;
-	uint64_t magic_version;
 	uint16_t gflags;
 	uint32_t cflags = htobe32(NBD_FLAG_FIXED_NEWSTYLE | NBD_FLAG_NO_ZEROES);
 	int sock = prv->socket;
-	char buffer[RECV_BUFFER_SIZE];
-
-	if (tdnbd_wait_read(sock) <= 0) {
-		ERROR("Timeout in nbd_negotiate");
-		close(sock);
-                return -1;
-        }
-
-        rc = recv(sock, buffer, 8, 0);
-        if (rc < 8) {
-		ERROR("Short read in negotiation(1) (%d)\n", rc);
-		close(sock);
-		return -1;
-        }
-
-	if (memcmp(buffer, "NBDMAGIC", 8) != 0) {
-	  buffer[8] = 0;
-	  ERROR("Error in NBD negotiation: got '%s'", buffer);
-	  close(sock);
-	  return -1;
-	}
-
-	if (tdnbd_wait_read(sock) <= 0) {
-	  ERROR("Timeout in nbd_negotiate");
-	  close(sock);
-	  return -1;
-	}
-
-	/* Receive NBD magic_version */
-	rc = recv_fully_or_fail(sock, &magic_version, sizeof(magic_version));
-	if (rc < 0)
-	{
-		ERROR("Failed to read magic from sock");
-		close(sock);
-		return -1;
-	}
-
-	if (ntohll(magic_version) != NBD_NEW_VERSION) {
-		ERROR("Not enough magic in negotiation(2) (%"PRIu64")\n",
-				ntohll(magic_version));
-		close(sock);
-		return -1;
-	}
 
 	/* Receive NBD flags */
-	rc = recv(sock, &gflags, sizeof(gflags), 0);
+	rc = tdnbd_wait_recv(sock, &gflags, sizeof(gflags), 0);
+	if (rc < 0) {
+		ERROR("Error in nbd_negotiate: %s", strerror(-rc));
+		goto errout;
+	}
 	if (rc < sizeof(gflags)) {
-		ERROR("Short read in negotiation(2) (%d)\n", rc);
-
-		return -1;
+		ERROR("Short read in NEW negotiation(3) (%d)\n", rc);
+		goto errout;
 	}
 
 	/* Send back flags*/
 	rc = send(sock, &cflags, sizeof(cflags), 0);
 	if (rc < sizeof(cflags)) {
 		ERROR("Failed to send client flags");
-		return -1;
+		goto errout;
 	}
 
 	return negotiate_client_newstyle_options(sock, driver);
+
+errout:
+	close(sock);
+	return -1;
 }
 
 static int
 tdnbd_nbd_negotiate(struct tdnbd_data *prv, td_driver_t *driver)
 {
-	if(CLIENT_USE_OLD_HANDSHAKE && !td_flag_test(driver->state, TD_DRIVER_NEW_NBD)) {
+	int rc;
+	uint64_t magic;
+	int sock = prv->socket;
+
+	/* Read the NBD opening magic number, which is the same for all protocol
+	 * versions */
+    rc = tdnbd_wait_recv(sock, &magic, sizeof(magic), 0);
```

Comment:
mixed tabs and spaces? and 747. (Yes, I know, we really need to run uncrustify on this code)